src/react/components/pages/editorPage/editorToolbar.tsx

```diff
@@ -17,6 +18,8 @@ export interface IEditorToolbarProps {
     project: IProject;
     actions: IProjectActions;
     items: IToolbarItemRegistration[];
+    canvas: Canvas;
+    onEditorModeChange: (mode: EditorMode) => void;
```

Comment:
Recommend renaming to "onItemSelected" and keeping the editor out of the toolbar as much as possible.  Then we can link directly into the `onToolbarItemSelected` method below to auto raise this event handler vs. each button having to implement it on its own.

Reply:
I think this would then allow us to reduce the coupling and `canvas` wouldn't be required anymore.  The editorPage itself can then listen for the `onItemSelected` event to then call into the canvas to set the selection mode.